engines/order_management/app/services/order_management/reports/enterprise_fee_summary/summarizer.rb

```diff
@@ -0,0 +1,80 @@
+module OrderManagement
+  module Reports
+    module EnterpriseFeeSummary
+      class Summarizer
+        attr_reader :data
+
+        delegate :fee_type, :enterprise_name, :fee_name, :fee_placement,
+                 :fee_calculated_on_transfer_through_name, :tax_category_name, to: :representation
+
+        def initialize(data)
+          @data = data
+        end
+
+        def customer_name
+          data["customer_name"]
+        end
+
+        def total_amount
+          data["total_amount"]
+        end
+
+        def i18n_translate(translation_key, options = {})
+          I18n.t("order_management.reports.enterprise_fee_summary.#{translation_key}", options)
+        end
+
+        private
+
+        def representation
+          @representation ||= representation_klass.new(self)
+        end
+
+        def representation_klass
+          return DataRepresentations::PaymentMethodFee if for_payment_method?
+          return DataRepresentations::ShippingMethodFee if for_shipping_method?
+          enterprise_fee_adjustment_presentation_klass if for_enterprise_fee?
+        end
+
+        def enterprise_fee_adjustment_presentation_klass
+          return DataRepresentations::CoordinatorFee if for_coordinator_fee?
```

Comment:
I used to write code this way, but the Ruby community seems to prefer another style:
```rb
if for_payment_method?
  DataRepresentations::PaymentMethodFee
elsif for_shipping_method?
  DataRepresentations::ShippingMethodFee
#...
end
```

Now, after the refactoring, there is a bit of duplication in the naming here. Everything is mentioned twice. If I wrote it from scratch, I would remove one level here:

```rb
if data["payment_method_name"].present?
  DataRepresentations::PaymentMethodFee
#...
```

But I don't think you need to touch this now. Just something to keep in mind when changing code in the future.